### ISSUE TITLE
Cursor not displaying over correct position

### ISSUE THREAD

**Issue by zacaj** [microsoft/vscode#69909]
Issue Type: <b>Bug</b>

Sometimes, my cursor, and any text highlights, aren't displayed in the correct position horizontally. 

I usually edit at zoom level 0 (default), and in that case the cursor may sometimes display to the right of where it should be.  If I use Ctrl+- to go down to level -1, the issue goes away.  

I can't reliably reproduce this issue, it seems to happen randomly after a period of time, possibly related to monitors being plugged+unplugged and that changing DPI settings?

It seems to match the issue described in https://github.com/Microsoft/vscode/issues/26151, but that issue seems to have been solved.  It references a 'disableMonospaceOptimizations' setting, but that setting seems to no longer exist?  

My fontsize is 12 and my font family is default (Consolas, 'Courier New', monospace)

VS Code version: Code 1.31.1 (1b8e8302e405050205e69b59abb3559592bb9e60, 2019-02-12T02:20:54.427Z)
OS version: Windows_NT x64 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2712)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|23.88GB (3.82GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (11)</summary>

Extension|Author (truncated)|Version
---|---|---
ng-template|Ang|0.1.11
path-intellisense|chr|1.4.2
bracket-pair-colorizer-2|Coe|0.0.28
EditorConfig|Edi|0.13.0
sass-lint|gle|1.0.4
vscode-jasmine-test-adapter|hbe|1.4.1
vscode-test-explorer|hbe|2.6.0
vscode-typescript-tslint-plugin|ms-|1.0.0
vsliveshare|ms-|0.3.1284
debugger-for-chrome|msj|4.11.3
vscode-subword-navigation|ow|1.2.0

</details>

**Comment by markgossa** [comment]
Hi,

Also have the same problem. Details of my setup are below:

Version: 1.33.1 (user setup)
Commit: 51b0b28134d51361cf996d2f0a1c698247aeabd8
Date: 2019-04-11T08:27:14.102Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.17134
Microsoft Surface Pro 4

If I click just to the left of a character, the cursor actually ends up to the right of the character or sometimes on top of the character. Occasionally it's the other way around when in fact the cursor ends up to the left of where I clicked.  

I found a workaround in that if I close and reopen VSCode then it's back to normal. If I have multiple instances of VSCode then I actually need to reopen each instance in turn. 

Potentially linked to my display setup or changing between different setups without restarts? At home I use 3xHD with the laptop screen off, at work I have a 4K screen with laptop display on.
